src/chess/chess_game.rs

```diff
@@ -180,6 +183,18 @@ fn chess_color_to_game_color(c: chess::Color) -> GameColor {
     }
 }
 
+impl PartialEq for ChessPosition {
+    fn eq(&self, other: &Self) -> bool {
+        self.board == other.board
+    }
+}
+impl Eq for ChessPosition {}
+impl Hash for ChessPosition {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.board.hash(state);

```

Comment:
do you not need the fifty_rule_count here? 
also can you change the name to `fifty` instead of `fifth`?